router/core/operation_planner.go

```diff
@@ -60,8 +61,8 @@ func (p *OperationPlanner) preparePlan(requestOperationName []byte, requestOpera
 
 	validation := astvalidation.DefaultOperationValidator()
 
-	// validate the document before planning
-	state := validation.Validate(&doc, p.executor.Definition, &report)
+	// validate the document against client schema before planning
+	state := validation.Validate(&doc, p.executor.ClientSchema, &report)
```

Comment:
Are we normalizing, etc... against the client schema as well? It would sound reasonable to me.

Reply:
yeah, everything except planning should be done against client schema